File: PerfTest1/PerfTest1-Xamarin/PerfTest1-ClassicXamarin.Android/MainActivity.cs
using System;
using System.Collections.Generic;
using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using PerfTest1Xamarin;
using System.Threading.Tasks;

namespace PerfTest1_ClassicXamarin.Android
{
    [Activity(Label = "PerfTest1_ClassicXamarin.Android", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity, View.IOnClickListener
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            if (this.ActionBar != null)
            {
                this.ActionBar.NavigationMode = ActionBarNavigationMode.Tabs;
            }

            SetContentView(Resource.Layout.activity_main);

            var tabListener = new MainTabListener();

            var tab = this.ActionBar.NewTab();
            tab.SetText("Azure List");
            tab.SetTabListener(tabListener);
            this.ActionBar.AddTab(tab);

            tab = this.ActionBar.NewTab();
            tab.SetText("Calc Primes");
            tab.SetTabListener(tabListener);
            this.ActionBar.AddTab(tab);

        }

        public void OnClick(View view)
        {
            if (view != null)
            {
                if (view.Id == Resource.Id.btnCalcPrimes)
                {
                    this.CalcPrimes();
                }
                else if (view.Id == Resource.Id.btnClear)
                {
                    this.ClearList();
                }
                else if (view.Id == Resource.Id.btnFetch)
                {
                    Task.Run(async () => await this.FillListAsync());
                }
            }
        }

        private async Task FillListAsync()
        {
            var registrations = await AzureTable.GetRegistrationsAsync();
            RunOnUiThread(() => ResetList(registrations));
        }

        private void ResetList(IList<Registration> registrations)
        {
            ListView lstRegistrations = (ListView)this.FindViewById(Resource.Id.lstRegistrations);
            if (lstRegistrations != null)
            {
                var adapter = ((RegistrationAdapter)lstRegistrations.Adapter);
                adapter.SetRegistrations(registrations);
                adapter.NotifyDataSetChanged();
                //lstRegistrations.RefreshDrawableState();
            }
        }

        private void ClearList()
        {
            ResetList(new List<Registration>());
        }

        private void CalcPrimes()
        {
            EditText txtMaxPrime = (EditText)this.FindViewById(Resource.Id.txtMaxPrime);

            int val = 0;
            if (int.TryParse(txtMaxPrime.Text, out val))
            {
                var returnValue = PerfTest1Xamarin.PrimeNumbers.GetPrimesFromSieve(val);

                AlertDialog.Builder dlgAlert = new AlertDialog.Builder(this);
                dlgAlert.SetMessage("Largest prime found: " + returnValue);
                dlgAlert.SetTitle("Prime Calculation Complete");
                dlgAlert.SetPositiveButton("OK", (sender, args) => {  });
                dlgAlert.SetCancelable(true);
                dlgAlert.Create().Show();
            }
            else
            {
                AlertDialog.Builder dlgAlert = new AlertDialog.Builder(this);
                dlgAlert.SetMessage(string.Format("{0}{1}", "Must enter a numeric max value: ", txtMaxPrime.Text));
                dlgAlert.SetTitle("Prime Calculation Error");
                dlgAlert.SetPositiveButton("OK", (sender, args) => { });
                dlgAlert.SetCancelable(true);
                dlgAlert.Create().Show();

            }            
        }
    }
}

File: PerfTest1/PerfTest1-Java/app/src/main/java/com/vandammeford/kevinf/perftest1_java/MainActivity.java
package com.vandammeford.kevinf.perftest1_java;

import android.app.ActionBar;
import android.app.Activity;
import android.app.AlertDialog;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ListView;

import com.microsoft.windowsazure.mobileservices.*;
import java.net.MalformedURLException;
import java.util.ArrayList;
import java.util.List;


public class MainActivity extends Activity implements View.OnClickListener  {

    private MobileServiceClient mobileServicesClient;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getActionBar() != null)
        {
            getActionBar().setNavigationMode(ActionBar.NAVIGATION_MODE_TABS);
        }
        setContentView(R.layout.activity_main);

        ActionBar.TabListener tabListener = new MainTabListener();

        ActionBar.Tab tab = getActionBar().newTab();
        tab.setText("Azure List");
        tab.setTabListener(tabListener);
        getActionBar().addTab(tab);

        tab = getActionBar().newTab();
        tab.setText("Calc Primes");
        tab.setTabListener(tabListener);
        getActionBar().addTab(tab);

        try {
            mobileServicesClient = new MobileServiceClient(
                    "https://malor2014jsmobileservice.azure-mobile.net/",
                    "pdFskoBXcwzaDNTpuRWdVRhUIRYcFF14",
                    this
            );
        } catch (MalformedURLException ex) {
            // Demo code, no need to handle.
        }
    }

    @Override
    public void onClick(View view) {
        if (view != null)
        {
            if (view.getId() == R.id.btnCalcPrimes)
            {
                this.calcPrimes();
            } else if (view.getId() == R.id.btnClear) {
                this.clearList();
            } else if (view.getId() == R.id.btnFetch) {
                this.fillList();
            }
        }
    }

    private void fillList() {

        mobileServicesClient.getTable(Registration.class).top(1000).execute(new TableQueryCallback<Registration>() {
               @Override
               public void onCompleted(List<Registration> result, int count, Exception exception,
                                       ServiceFilterResponse response) {
             final List<Registration> registrations = result;
             runOnUiThread(new Runnable() {
                       @Override
                       public void run() {
                           resetList(registrations);
                       }
                   });
               }
            }
        );
    }

    private void resetList(List<Registration> registrations) {
        ListView lstRegistrations = (ListView)this.findViewById(R.id.lstRegistrations);
        if (lstRegistrations != null) {
            ((RegistrationAdapter)lstRegistrations.getAdapter()).setRegistrations(registrations);
            ((RegistrationAdapter) lstRegistrations.getAdapter()).notifyDataSetChanged();
        }
    }

    private void clearList() {
        resetList(new ArrayList<Registration>());
    }

    private void calcPrimes() {
        EditText txtMaxPrime = (EditText)this.findViewById(R.id.txtMaxPrime);
        try
        {
            int maxValue = Integer.parseInt(txtMaxPrime.getText().toString());
            int maxPrime = getPrimesFromSieve(maxValue);

            AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
            dlgAlert.setMessage("Largest prime found: " + maxPrime);
            dlgAlert.setTitle("Prime Calculation Complete");
            dlgAlert.setPositiveButton("OK", null);
            dlgAlert.setCancelable(true);
            dlgAlert.create().show();                }
        catch(NumberFormatException nfe)
        {
            AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
            dlgAlert.setMessage("Must enter a numeric max value: " + txtMaxPrime.getText().toString());
            dlgAlert.setTitle("Prime Calculation Error");
            dlgAlert.setPositiveButton("OK", null);
            dlgAlert.setCancelable(true);
            dlgAlert.create().show();
        }
    }

    private int getPrimesFromSieve(int maxValue)
    {
        byte[] primes = new byte[maxValue + 1];
        for (int i = 0; i <=maxValue; i++)
        {
            primes[i] = 0;
        }
        int largestPrimeFound = 1;

        for (int i = 2; i <=maxValue; i++)
        {
            if (primes[i - 1] == 0)
            {
                primes[i - 1] = 1;
                largestPrimeFound = i;
            }

            int c = 2;
            int mul = i*c;
            for (; mul <= maxValue;)
            {
                primes[mul - 1] = 1;
                c++;
                mul = i*c;
            }
        }
        return largestPrimeFound;
    }
}
